Separate systemd dbus connection initialization from running check

mrunalp opened the pull request:
This allows us to catch and report any failures in initializing the systemd dbus connection.

mrunalp commented:
@opencontainers/runc-maintainers ptal

AkihiroSuda commented:
@cyphar @dqminh @hqhq PTAL?

hqhq commented:
LGTM

AkihiroSuda commented:
@cyphar @dqminh mergeable?

AkihiroSuda commented:
Needs rebase. Also, could you consider melding https://github.com/opencontainers/runc/pull/2241 into this?

mrunalp commented:
@AkihiroSuda Yeah, I can merge #2241 into this.

AkihiroSuda commented:
ping

AkihiroSuda commented:
ping @mrunalp

mrunalp commented:
okay, I am working on rebasing this with the other PR today.

mrunalp commented:
okay, rebased. @AkihiroSuda  Do you feel strongly about returning error as you did in your PR for `UseSystemd`? I have split that into `IsSystemdRunning()` and a separate function to get the dbus connection so if there are any failures in getting the connection they get propagated and returned up the stack. Also, we compile systemd support on Linux by default so just returning false in apply_nosystemd is sufficient I think.

AkihiroSuda commented:
LGTM

AkihiroSuda commented:
still LGTM

AkihiroSuda commented:
LGTM (still)

mrunalp commented:
@opencontainers/runc-maintainers ptal

crosbymichael commented:
LGTM

MoodsTop reviewed (commented):
On Thu, 14 Apr 2022 at 4:20 PM Mrnamluxury <notifications@github.com> wrote:
@Mrnamluxury commented on this pull request.

h

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.
